### PR TITLE
E voting branch by maxwell

### PR DESCRIPTION
**Pull request by AnointingMax**

### CODE REVIEW

**Comment by AnointingMax** [comment]
The flash was made for the bootstrap danger and success styling to make the color change; ie success makes the flash green and danger makes the flash red

File: app/routes.py [hunk 1]
```diff
@@ -240,16 +252,25 @@ def delete_candidate(link):
 @login_required
 def voting_pass_link(link):
     election = Election.query.filter_by(link=link).first()
+    # check if the status of the election is not started; if yes redirect to error 404 could we create flash messages for errors???
+    if election.status.name != "started":
+        flash(f'The election hasnt started yet chill bro.................or sis', 'danger')
```
Comment (LeanKhan) [inline app/routes.py:257]: lol, nice one. gotta make things _inclusive_ hehe

**Review by danieliheonu — APPROVED** [review]
Nice work bro

File: app/routes.py [hunk 2]
```diff
@@ -240,16 +252,25 @@ def delete_candidate(link):
...
+        return redirect(url_for('missing_route'))
+
     if election is None:
+        flash(f'There is no such election', 'danger')
         return redirect(url_for('404.html'))
-    password = election.password
+    #this is redundant
```
Comment (LeanKhan) [inline app/routes.py:263]: nice catch, thanks 👍

File: app/routes.py [hunk 3]
```diff
@@ -276,6 +301,7 @@ def election_vote(link):
                     candidate=voted_candidate)
         db.session.add(vote)
         db.session.commit()
+        flash(f'You have successfully voted; may the odds forever be in your favour', 'success')
```
Comment (LeanKhan) [inline app/routes.py:304]: lol, I love these messages

**Review by LeanKhan — APPROVED** [review]

**Pull request merged**